### ISSUE TITLE
Switch development to a `develop` branch.

### ISSUE THREAD

**Issue by djbe** [SwiftGen/SwiftGen#432]
Currently all main development happens on `master`, but this leads to confusion such as in #431, where the documentation has already been updated with annotations for breaking changes, which only confuses users of the current release.

I propose adding a `develop` branch, and have all PRs merge into that branch. Once a release is ready, merge it into `master` and tag it.

**Comment by AliSoftware** [comment]
I like the idea of using git-flow instead of github-flow.

The only drawback is that when creating a PR we'd have to not forget about changing the target branch to develop. Even is could forget sometimes, but also this means other Contributors have to be aware of that

Idk if GitHub has the concept of default branches for PRs without making that branch also the default branch displayed when landing on the repo'sb home page? Worth investigating

**Comment by djbe** [comment]
We could try changing the "default branch" in the settings to `develop` and see what happens?

**Comment by AliSoftware** [comment]
I just tried, by creating a `develop` branch, and changing the default branch in the settings to `develop`. Then when I go back to https://github.com/SwiftGen/SwiftGen 's home page, `branch: develop` gets selected and shown by default — meaning that the `develop`'s README is the one shown by default too.

(I've removed the `develop` branch since that test, to avoid confusion for now)

**Comment by AliSoftware** [comment]
To be honest, none of the other popular OpenSource tools I know, e.g. SwiftLint, Fastlane, Sourcery, etc all use the same flow as we do, i.e. GitFlow, and all behave the same: the landing page shows `master`, and their shield.io badge shows the latest released (podspec) version, meaning that exactly like for SwiftGen, the README shown to users by default (the one in `master`) doesn't necessary match the documentation for the latest released version, while the shields.io badge still shows the latest released version on top of that README.

I think a better solution, rather than switching to gitflow which seems to have at least as many drawbacks than advantages (or maybe less drawbacks for users but way more drawbacks for contributors, since it seems we can't have a different default branch for PR and for landing page), would be to have a jazzy-generated (or any other tool like jekyll or whatnot) documentation WebSite.

We actually already have a repo that was created exactly to host such a documentation website that we intended to create one day, but never had the time to tackle this step so far. I think that might be the best solution anyway. And that's also the road that both Stencil and Sourcery already have taken, so…

**Comment by djbe** [comment]
Closing this in favour of https://github.com/SwiftGen/swiftgen.github.io/issues/1.
